src/qibo/transpiler/pipeline.py

```diff
@@ -0,0 +1,221 @@
+            np.testing.assert_allclose(fidelity, 1.0)
+        except AssertionError:
+            raise_error(TranspilerPipelineError, "Circuit equivalence not satisfied.")
+
+
+def _transpose_qubits(state: np.ndarray, qubits_ordering: np.ndarray):
+    """Reorders qubits of a given state vector.
+
+    Args:
+        state (np.ndarray): final state of the circuit.
+        qubits_ordering (np.ndarray): final qubit ordering.
+    """
+    original_shape = state.shape
+    state = np.reshape(state, len(qubits_ordering) * (2,))
+    state = np.transpose(state, qubits_ordering)
+    return np.reshape(state, original_shape)
+
+
+def assert_transpiling(
+    original_circuit: Circuit,
+    transpiled_circuit: Circuit,
+    connectivity: nx.Graph,
+    initial_layout: dict,
+    final_layout: dict,
+    native_gates: NativeType = NativeType.CZ,
+    check_circuit_equivalence=True,
+):
+    """Check that all transpiler passes have been executed correctly.
+
+    Args:
+        original_circuit (qibo.models.Circuit): circuit before transpiling.
+        transpiled_circuit (qibo.models.Circuit): circuit after transpiling.
+        connectivity (networkx.Graph): chip qubits connectivity.
+        initial_layout (dict): initial physical-logical qubit mapping.
+        final_layout (dict): final physical-logical qubit mapping.
+        native_gates: (NativeType): native gates supported by the hardware.
+    """
+    assert_connectivity(circuit=transpiled_circuit, connectivity=connectivity)
+    assert_decomposition(circuit=transpiled_circuit, two_qubit_natives=native_gates)
+    if original_circuit.nqubits != transpiled_circuit.nqubits:
+        qubit_matcher = Preprocessing(connectivity=connectivity)
+        original_circuit = qubit_matcher(circuit=original_circuit)
+    assert_placement(circuit=original_circuit, layout=initial_layout)
+    assert_placement(circuit=transpiled_circuit, layout=final_layout)
+    if check_circuit_equivalence:
+        assert_circuit_equivalence(
+            original_circuit=original_circuit,
+            transpiled_circuit=transpiled_circuit,
+            initial_map=initial_layout,
+            final_map=final_layout,
+        )
+
+
+class Passes:
+    """Define a transpiler pipeline consisting of smaller transpiler steps that are applied sequentially:
+
+    Args:
+        passes (list): list of passes to be applied sequentially,
+            if None default transpiler will be used, it requires hardware connectivity.
+        connectivity (nx.Graph): hardware qubit connectivity.
+    """
+
+    def __init__(
+        self,
+        passes: list = None,
+        connectivity: nx.Graph = None,
+        native_gates: NativeType = NativeType.CZ,
+    ):
+        self.native_gates = native_gates
+        if passes is None:
+            self.passes = self.default(connectivity)
+        else:
+            self.passes = passes
+        self.connectivity = connectivity
+
+    def default(self, connectivity: nx.Graph):
```

Comment:
I would probably use as default here the best pipeline we have. Therefore, `ReverseTraversal` + `Sabre` I believe.

Reply:
Same as before, sabre needs the measurement fix.